winlogbeat/eventlog/wineventlog.go

```diff
@@ -317,7 +317,9 @@ func (l *winEventLog) Open(state checkpoint.EventLogState) error {
 	var err error
 	// we need to defer metrics initialization since when the event log
 	// is used from winlog input it would register it twice due to CheckConfig calls
-	l.metrics = newInputMetrics(l.channelName, l.id)
+	if l.metrics == nil {
+		l.metrics = newInputMetrics(l.channelName, l.id)
+	}
```

Comment:
does it mean that `l.Open` is called twice without `l.Close`? If so, there seems to be more issues

https://learn.microsoft.com/en-us/windows/win32/api/winevt/nf-winevt-evtsubscribe#return-value

so perhaps
```diff
-return l.openChannel(bookmark)
+ if l.subscription {
+    win.Close(l.subscription)
+}
+return l.openChannel(bookmark)
```

Reply:
It only gets called more than once in case it fails to open, so no need to call close in that case.